application/controllers/gcp_survey_response.py

```diff
@@ -0,0 +1,219 @@
+import hashlib
+import json
+import logging
+import sys
+import time
+import uuid
+
+import structlog
+from google.cloud import storage, pubsub_v1
+from google.cloud.exceptions import GoogleCloudError
+
+from application.controllers.helper import convert_file_object_to_string_base64
+from application.controllers.rabbit_helper import _encrypt_message
+from application.controllers.service_helper import (get_business_party, get_case_group, get_collection_exercise,
+                                                    get_survey_ref)
+log = structlog.wrap_logger(logging.getLogger(__name__))
+
+
+class FileTooSmallError(Exception):
+    pass
+
+
+class SurveyResponseError(Exception):
+    pass
+
+
+class GcpSurveyResponse:
+    def __init__(self, config):
+        self.config = config
+
+        # Bucket config
+        self.storage_client = None
+        self.seft_bucket_name = self.config['SEFT_BUCKET_NAME']
+        self.seft_bucket_file_prefix = self.config.get('SEFT_BUCKET_FILE_PREFIX')
+
+        # Pubsub config
+        self.publisher = None
+        self.gcp_project_id = self.config['GOOGLE_CLOUD_PROJECT']
+        self.seft_pubsub_topic = self.config['SEFT_PUBSUB_TOPIC']
+
+    """
+    The survey response from a respondent
+    """
+    def add_survey_response(self, case_id: str, file, file_name: str, survey_ref: str):
+        """
+        Encrypt and upload survey response to gcp bucket, and put metadata about it in pubsub.
+
+        :param case_id: A case id
+        :param file: A file object from which we can read the file contents
```

Comment:
Isn't this actually the file contents now?

Reply:
Done